src/components/auth-module/register-view/register-view.js

```diff
@@ -0,0 +1,241 @@
+import React, { Component } from 'react';
+import { translate } from 'react-i18next';
+import { translateOptions } from '../../../i18n/config';
+import { Button, Form } from 'semantic-ui-react';
+import { Link } from 'react-router-dom';
+
+import ErrorMessageView from '../../common/error-message-view';
+import { isValidEmail } from '../../../utils/validation-utils';
+import RegisterMessageView from './register-message/register-message';
+
+class RegisterView extends Component {
+  /***************************
+   *        CONSTRUCTOR
+   ***************************/
+  constructor(props) {
+    super(props);
+
+    this.state={
+      fields: {
+        email: '',
+        firstName: '',
+        lastName: '',
+        phoneNumber: '',
+        companyName: '',
+        password: '',
+      },
+      serverError: "",
+      errors: {}
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleFormValidation = this.handleFormValidation.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  /***************************
+   *          METHODS
+   ***************************/
+  handleChange(e){
+    let fields = this.state.fields;
+    const field = e.target.name;
+    fields[field] = e.target.value;
+    this.setState({
+      fields,
+      errors: {},
+      serverError: ""
+    })
+  }
+
+  handleSubmit = (e)=> {
+    e.preventDefault();
+    const formValidationFeedback = this.handleFormValidation(this.state.fields);
+    if(formValidationFeedback.isFormValid){
+      // API call for register request
+      this.setState({showErrorMessage: true});
+    }else{
+      this.setState({errors:formValidationFeedback.errors});
+    }
+  }
+
+  handleFormValidation(fields){
+    let errors = {};
+    let formIsValid = true;
+    let result = {};
+
+    if(fields['firstName'] === ''){
+      formIsValid = false;
+      errors["firstName"] = "*This field is required.";
+    }
+    if(fields['email'] === ''){
+      formIsValid = false;
+      errors["email"] = "*This field is required.";
+    } else if(typeof fields["email"] !== "undefined"){
```

Comment:
`undefined` should not be in the string. I have done the same mistake, I will also correct in the login view.